src/main/java/uk/gov/companieshouse/api/testdata/service/impl/CertificatesServiceImpl.java

```diff
@@ -37,29 +38,37 @@ public class CertificatesServiceImpl implements DataService<CertificatesData, Ce
     public RandomService randomService;
 
     @Override
-    public CertificatesData create(CertificatesSpec certificatesSpec) throws DataException {
-        Long firstPart = randomService.getNumber(6);
-        Long secondPart = randomService.getNumber(6);
-        var randomId = "CRT-" + firstPart + "-" + secondPart;
-        var certificates = getCertificates(certificatesSpec, randomId);
-
-        certificatesRepository.save(certificates);
+    public CertificatesData create(CertificatesSpec spec) throws DataException {
+        List<ItemOptionsSpec> optionsList = spec.getItemOptions();
+        List<CertificatesData.CertificateEntry> certificateEntries = new ArrayList<>();
```

Comment:
As it looks like the amount of elements in `certificateEntries` will always match the amount of elements in `optionsList`.

If that's the case then you should give the `ArrayList` the same initial capacity as it will stop it resizing or being too big than it needs to be.

```suggestion
        List<CertificatesData.CertificateEntry> certificateEntries = new ArrayList<>(optionsList.size());
```